astrodb_utils/spectra.py

```diff
@@ -0,0 +1,169 @@
+import importlib.util
+import logging
+import sys
+
+import astropy.units as u
+import numpy as np
+from specutils import Spectrum1D
+
+from astrodb_utils import AstroDBError
+
+matplotlib_check = importlib.util.find_spec("matplotlib")
+if matplotlib_check is not None:
+    import matplotlib.pyplot as plt
+
+
+__all__ = [
+    "check_spectrum_class", 
+    "check_spectrum_not_nans", 
+    "check_spectrum_wave_units", 
+    "check_spectrum_flux_units", 
+    "check_spectrum_plottable"
+    ]
+
+logger = logging.getLogger(__name__)
+
+
+def check_spectrum_class(spectrum, raise_error=True):
+    try:
+        Spectrum1D.read(spectrum)
+        return True
+    except Exception as error_message:
+        msg = f"Unable to load file as Spectrum1D object:{spectrum}"
+        logger.debug(f"{error_message}")
+        if raise_error:
+            logger.error(msg)
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return False
+
+
+def check_spectrum_not_nans(spectrum, raise_error=True):
```

Comment:
so it's OK to have some NaNs in the spectrum, as long as it's not ALL NaNs?

Reply:
That's correct. This criteria came from @Will-Cooper and the website. I'm not sure where spectra that's all NaNs came from but apparently it can be a thing.